core/resp.go

```diff
@@ -77,6 +77,11 @@ func readBulkString(c io.ReadWriter, buf *bytes.Buffer) (string, error) {
 		return "", err
 	}
 
+	//handling RESP_NIL case
+	if len == -1 {
+		return "(nil)", nil
```

Comment:
While this works, it might be more idiomatic to return an empty string and let callers handle the nil case.
```suggestion
		return "", nil
```

Reply:
Currently, it is called by `(*RESParser).DecodeOne()` which directly returns the result of `readBulkString`.

If we modify the code, it will be:

```go
...
switch b {
	case '+':
		return readSimpleString(rp.c, rp.buf)
	case '-':
		return readError(rp.c, rp.buf)
	case ':':
		return readInt64(rp.c, rp.buf)
	case '$':
		resString, err := readBulkString(rp.c, rp.buf)
		if resString == "" && err == nil {
			return "(nil)", err
		}
		return resString, err
	case '*':
		return readArray(rp.c, rp.buf, rp)
	}
...
```

Will this be fine?

Reply:
Maybe a better idea is to return `nil,nil` instead of `"", nil` for RESP_NIL case inside `readBulkString`.

Then we can change the DecodeOne call to directly return readBulkString.
```go
...
switch b {
	case '+':
		return readSimpleString(rp.c, rp.buf)
	case '-':
		return readError(rp.c, rp.buf)
	case ':':
		return readInt64(rp.c, rp.buf)
	case '$':
		return readBulkString(rp.c, rp.buf)
	case '*':
		return readArray(rp.c, rp.buf, rp)
	}
...
```

I would imagine that it would be the client's job to interpret the `nil` as `(nil)`, at the same time our codebase can expect the golang `nil` value.
```go
{
	InCmds: []string{"SET k v EX 1", "SLEEP 2", "GET k"},
	Out:    []interface{}{"OK", "OK", nil},
},
```

Reply:
@arpitbbhayani what are your thoughts around this?

Reply:
Resolving this for now, merging it since this fixes our currently broken tests.